kasa/httpclient.py

```diff
@@ -89,6 +102,8 @@ async def post(
                         response_data = json_loads(response_data.decode())
 
         except (aiohttp.ServerDisconnectedError, aiohttp.ClientOSError) as ex:
+            if isinstance(ex, aiohttp.ClientOSError):
+                self._wait_between_requests = self.WAIT_TIME
```

Comment:
The issue is that the error will cause the protocol to retry the whole request again so it will error again. N.B If it was trying to get incomplete lists it would have the same problem.